### PR TITLE
MAGECLOUD-4165: Networking Bridge Mode

### PR DESCRIPTION
**Pull request by BaDos**
### Fixed Issues (if relevant)

https://magento2.atlassian.net/browse/MAGECLOUD-4165

### Manual testing scenarios

Update composer.json before test:
```
    "repositories": {
        "ece-tools": {
            "type": "vcs",
            "url": "https://github.com/magento/ece-tools.git"
        },
        "docker": {
            "type": "vcs",
            "url": "https://github.com/magento-thunder/magento-cloud-docker.git"
        },
        "repo": {
            "type": "composer",
            "url": "https://repo.magento.com"
        }
    },
    "require": {
        "magento/magento-cloud-metapackage": ">=2.2.9 <2.2.10",
        "magento/ece-tools": "dev-MAGECLOUD-4004 as 2002.0.33",
        "magento/magento-cloud-docker": "dev-MAGECLOUD-4165 as 1.0.1"
    },
```
https://jira.corp.magento.com/browse/MAGECLOUD-116
https://jira.corp.magento.com/browse/MAGECLOUD-66

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages

### CODE REVIEW

**Review by shiftedreality — APPROVED** [review]

**Comment by BaDos** [comment]
Please, do not merge this PR.
I will let you know when you can merge it.

**Review by bbatsche — APPROVED** [review]
Something for our backlog:
I believe this is another value that will prevent running multiple instances of Magento Cloud Docker on a single machine. We should begin planning on how we can address this in a future update (perhaps a short hash of the host's path)

File: src/Command/Image/GeneratePhp.php [hunk 1]
```diff
@@ -263,16 +267,13 @@ private function buildDockerfile(string $dockerfile, string $phpVersion, string
...
                 'app-etc' => [
                     'def' => 'VOLUME ${MAGENTO_ROOT}/app/etc',
-                    'cmd' => 'RUN mkdir -p ${MAGENTO_ROOT}/app/etc && chown -R www:www ${MAGENTO_ROOT}/app/etc'
+                    'cmd' => 'RUN mkdir -p ${MAGENTO_ROOT}/app/etc && chown -R www:www ${MAGENTO_ROOT}/app'
                 ],
-                'pub-static' => [
-                    'def' => 'VOLUME ${MAGENTO_ROOT}/pub/static',
-                    'cmd' => 'RUN mkdir -p ${MAGENTO_ROOT}/pub/static && chown -R www:www ${MAGENTO_ROOT}/pub/static'
+                'pub-static-and-media' => [
+                    'def' => 'VOLUME ${MAGENTO_ROOT}/pub/static' . "\n" . 'VOLUME ${MAGENTO_ROOT}/pub/media',
```
Comment (shiftedreality) [inline src/Command/Image/GeneratePhp.php:273]: This can be combined

File: images/php/7.0-cli/Dockerfile [hunk 2]
```diff
@@ -200,14 +200,15 @@ RUN ["chmod", "+x", \
...
 RUN mkdir ${MAGENTO_ROOT} && chown -R www:www ${MAGENTO_ROOT}
+RUN mkdir ${MAGENTO_ROOT}/.composer && chown -R www:www ${MAGENTO_ROOT}/.composer
 RUN mkdir ${MAGENTO_ROOT}/vendor && chown -R www:www ${MAGENTO_ROOT}/vendor
 RUN mkdir ${MAGENTO_ROOT}/generated && chown -R www:www ${MAGENTO_ROOT}/generated
 RUN mkdir ${MAGENTO_ROOT}/var && chown -R www:www ${MAGENTO_ROOT}/var
-RUN mkdir -p ${MAGENTO_ROOT}/app/etc && chown -R www:www ${MAGENTO_ROOT}/app/etc
-RUN mkdir -p ${MAGENTO_ROOT}/pub/static && chown -R www:www ${MAGENTO_ROOT}/pub/static
-RUN mkdir -p ${MAGENTO_ROOT}/pub/media && chown -R www:www ${MAGENTO_ROOT}/pub/media
+RUN mkdir -p ${MAGENTO_ROOT}/app/etc && chown -R www:www ${MAGENTO_ROOT}/app
+RUN mkdir -p ${MAGENTO_ROOT}/pub/static && mkdir -p ${MAGENTO_ROOT}/pub/media && chown -R www:www ${MAGENTO_ROOT}/pub
```
Comment (bbatsche) [inline images/php/7.0-cli/Dockerfile:208]: This whole chunk would be better as a single `RUN` with `mkdir ... & mkdir ... & chown ... & chown ... etc`, but it's not necessary for this PR

**Review by shiftedreality — APPROVED** [review]

**Review by bbatsche — APPROVED** [review]

**Comment by andriyShevtsov** [comment]
QA approved

**Pull request merged**